// Copyright 2018 The SAF Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

message Feature {
    repeated double feature = 1;
}

message DetectionProto {
    message Thumbnail {
        optional bytes thumbnail = 1;
        required string label = 2;
        optional Feature feature = 3;
        optional string id = 4;
    }
    required string capture_time_micros = 1;
    required string stream_id = 2;
    repeated Thumbnail thumbnails = 3;
    required uint64 frame_id = 4;
}

message FrameProto {
    message Rect {
        required int32 x = 1;
        required int32 y = 2;
        required int32 w = 3;
        required int32 h = 4;
    }
    message RectInfo {
        required Rect bbox = 1;
        required string label = 2;
        optional Feature feature = 3;
        optional string id = 4;
    }
    required string capture_time_micros = 1;
    required string stream_id = 2;
    optional bytes image = 3;
    repeated RectInfo rect_infos = 4;
    required uint64 frame_id = 5;
}
